hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCacheManager.java

```diff
@@ -47,8 +48,7 @@
     private final Object mutex = new Object();
     private final Queue<ScheduledFuture> preloadTaskFutures = new ConcurrentLinkedQueue<>();
     private final ConcurrentMap<String, NearCache> nearCacheMap = new ConcurrentHashMap<>();
-
-    private volatile ScheduledFuture storageTaskFuture;
+    private final AtomicReference<Object> storageTaskFutureRef = new AtomicReference<>();
```

Comment:
`AtomicReference<ScheduledFuture>` ?

Reply:
Using Object is on purpose. Below CAS operation puts `this` reference.
```
private void createAndScheduleStorageTask() {
        if (storageTaskFutureRef.compareAndSet(null, this)) {
            storageTaskFutureRef.set(scheduler.scheduleWithRepetition(new StorageTask(),
                    0, 1, SECONDS));
        }
    }
```